internal/dot/graph.go

```diff
@@ -190,6 +203,7 @@ func (dg *Graph) AddMissingNodes(results []*Result) {
 func (dg *Graph) FailNodes(results []*Result, id CtorID) {
 	// This failure is the root cause if there are no other failures.
 	isRootCause := len(dg.Failed.RootCauses) == 0
+	dg.Failed.FailedCtors[id] = struct{}{}
```

Comment:
notice the stuttering here: `dg.Failed.FailedCtors`. We're already inside FailedNodes so "failed" is already in context. The names of the fields inside FailedNodes don't need to include the word "failed".